Sources/CovidCertificateSDK/CertLogic/CertLogic.swift

```diff
@@ -126,11 +128,27 @@ class CertLogic {
                                            isSwitzerlandOnly: isSwitzerlandOnly))
     }
 
-    private func createPayload(from holder: CertificateHolderType) -> CertLogicPayload? {
-        guard let certificate = holder.certificate as? DCCCert else {
-            return nil
+    func checkModeRules(holder: CertificateHolderType, modes: [CheckMode], validationClock: Date = Date()) -> Result<ModeResults, CertLogicValidationError> {
+        var results: [CheckMode: ModeCheckResult] = [:]
+        let external = externalJson(validationClock: validationClock)
+        for mode in modes {
+            guard let modeRule = modeRule,
+                  let payload = createPayload(from: holder, mode: mode.id),
+                  let json = try? JSONEncoder().encode(payload) else {
+                return .failure(.JSON_ERROR)
```

Comment:
Same here, we're failing the whole `checkModeRules`, instead of marking the result of this mode as `.failure`